server/datastore/cached_mysql/cached_mysql.go

```diff
@@ -228,15 +232,35 @@ func (ds *cachedMysql) TeamAgentOptions(ctx context.Context, teamID uint) (*json
 	return agentOptions, nil
 }
 
+func (ds *cachedMysql) TeamFeatures(ctx context.Context, teamID uint) (*fleet.Features, error) {
+	key := fmt.Sprintf(teamFeaturesKey, teamID)
+	if x, found := ds.c.Get(key); found {
+		if features, ok := x.(*fleet.Features); ok {
+			return features, nil
+		}
+	}
+
+	features, err := ds.Datastore.TeamFeatures(ctx, teamID)
+	if err != nil {
+		return nil, err
+	}
+
+	ds.c.Set(key, features, ds.teamFeaturesExp)
+
+	return features, nil
+}
+
 func (ds *cachedMysql) SaveTeam(ctx context.Context, team *fleet.Team) (*fleet.Team, error) {
 	team, err := ds.Datastore.SaveTeam(ctx, team)
 	if err != nil {
 		return nil, err
 	}
 
-	key := fmt.Sprintf(teamAgentOptionsKey, team.ID)
+	agentOptionsKey := fmt.Sprintf(teamAgentOptionsKey, team.ID)
+	featuresKey := fmt.Sprintf(teamFeaturesKey, team.ID)
 
-	ds.c.Set(key, team.Config.AgentOptions, ds.teamAgentOptionsExp)
+	ds.c.Set(agentOptionsKey, team.Config.AgentOptions, ds.teamAgentOptionsExp)
+	ds.c.Set(featuresKey, &team.Config.Features, ds.teamFeaturesExp)
```

Comment:
Why a pointer? Curious.

Reply:
Also following the other methods in this file, especially those related to `AppConfig` since they deal with the same kind of data.